### PR TITLE
1046 | Consuming link notification from queue. 

### PR DESCRIPTION
**Pull request by angshu**

### CODE REVIEW

File: src/main/java/in/projecteka/datanotificationsubscription/hipLink/HipLinkNotificationListener.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        var mlc = messageListenerContainerFactory.createMessageListenerContainer(HIP_LINK_QUEUE);
+        MessageListener messageListener = message -> {
+            try {
+                logger.info("Message from queue: {} ", message);
+                ObjectMapper mapper = new ObjectMapper();
+                mapper.registerModule(new JavaTimeModule());
+                TraceableMessage traceableMessage = mapper.readValue(new String(message.getBody(), "UTF-8"), TraceableMessage.class);
+                NewCCLinkEvent hipLinkEvent = mapper.convertValue(traceableMessage.getMessage(), NewCCLinkEvent.class);
+                logger.debug("Received Link Event message {}", hipLinkEvent);
+                subscriptionManager.notifySubscribers(hipLinkEvent);
```
Comment (Nexengineer) [inline src/main/java/in/projecteka/datanotificationsubscription/hipLink/HipLinkNotificationListener.java:37]: Please extract this whole in a new function and wrap it in a try-catch block.

Comment (angshu) [reply src/main/java/in/projecteka/datanotificationsubscription/hipLink/HipLinkNotificationListener.java:37]: I deliberately removed the function. I will add the try catch to log the error

Comment (angshu) [reply src/main/java/in/projecteka/datanotificationsubscription/hipLink/HipLinkNotificationListener.java:37]: actually the try - catch is already there

Comment (Nexengineer) [reply src/main/java/in/projecteka/datanotificationsubscription/hipLink/HipLinkNotificationListener.java:37]: The problem is, I have seen if your API call fails. It doesn't fail gracefully and this piece of code will execute infinite times.Keeping it in seprate method works. It is DLQ problem.

Comment (angshu) [reply src/main/java/in/projecteka/datanotificationsubscription/hipLink/HipLinkNotificationListener.java:37]: it throws new AmqpRejectAndDontRequeueException()

Comment (Nexengineer) [reply src/main/java/in/projecteka/datanotificationsubscription/hipLink/HipLinkNotificationListener.java:37]: Doing operations like saving in db or calling an api inside subscribe function make it not work properly

**Review by Nexengineer — APPROVED** [review]

**Pull request merged**